### PR TITLE
Monolinear 1.1 65k

### PR DESCRIPTION
**Pull request by M0N0S0DIUM**
Monolinear 1.1 initial release

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/monolinear11/VIA.json
Comment (fauxpark) [inline keyboards/monolinear11/VIA.json:1]: VIA JSON is not allowed in this repo. Please remove it.

File: keyboards/monolinear11/config.h
Comment (fauxpark) [inline keyboards/monolinear11/config.h:1]: All of these defines can be migrated to info.json.

File: keyboards/monolinear11/info.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
+{
+    "keyboard_name": "Monolinear_1.1",
+    "manufacturer": "Jacob",
+    "url": "N/A",
+    "maintainer": "Vial",
```
Comment (fauxpark) [inline keyboards/monolinear11/info.json:5]: No.

File: keyboards/monolinear11/info.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    "usb": {
+        "vid": "0xA454",
+        "pid": "0x0000",
+        "device_version": "1.0.1"
+    },
+    "rgblight": {
+        "max_brightness": 100
+    },
```
Comment (fauxpark) [inline keyboards/monolinear11/info.json:13]:
```suggestion
```
RGB Matrix is enabled, not RGBLight.

File: keyboards/monolinear11/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    [1] = LAYOUT(
+        _______, _______, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_TRANSPARENT,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_MOD, _______,
+        DB_TOGG, _______, _______, _______, _______, RGB_VAD, RGB_RMOD, RGB_VAI
+    )
+};
+
+led_config_t g_led_config  = {
```
Comment (fauxpark) [inline keyboards/monolinear11/keymaps/default/keymap.c:23]: `g_led_config` should not live in the keymap.

File: keyboards/monolinear11/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    [1] = LAYOUT(
+        _______, _______, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_TRANSPARENT,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_MOD, _______,
+        DB_TOGG, _______, _______, _______, _______, RGB_VAD, RGB_RMOD, RGB_VAI
+    )
+};
+
+led_config_t g_led_config  = {
```
Comment (fauxpark) [inline keyboards/monolinear11/keymaps/via/keymap.c:23]: As above.

File: keyboards/monolinear11/rules.mk [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = yes            # Link time optimization
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = ws2812
```
Comment (fauxpark) [inline keyboards/monolinear11/rules.mk:12]: All of this can be migrated to info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
Readme is also missing.

File: keyboards/monolinear11/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        KC_ESC,  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,  TG(1),
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_UP,   KC_DEL,
+        KC_LCTL, KC_LGUI, KC_LALT,                                 KC_SPC,                       KC_LALT, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT(
+        _______, _______, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_TRANSPARENT,
```
Comment (waffle87) [inline keyboards/monolinear11/keymaps/default/keymap.c:17]:
```suggestion
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

File: keyboards/monolinear11/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        KC_ESC,  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,  TG(1),
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_UP,   KC_DEL,
+        KC_LCTL, KC_LGUI, KC_LALT,                                 KC_SPC,                       KC_LALT, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT(
+        _______, _______, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_TRANSPARENT,
```
Comment (waffle87) [inline keyboards/monolinear11/keymaps/via/keymap.c:17]:
```suggestion
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

File: keyboards/monolinear11/VIA.json
Comment (M0N0S0DIUM) [reply keyboards/monolinear11/VIA.json:1]: understood.

**Pull request closed without merging**